src/egon/data/processing/heat_demand_timeseries/HTS.py

```diff
@@ -0,0 +1,1203 @@
+
+        mv_grid = psycop_df_AF('boundaries.egon_map_zensus_grid_districts')
+        mv_grid = mv_grid.set_index('zensus_population_id')
+        district_heating =district_heating.set_index('zensus_population_id')
+        mv_grid_ind = mv_grid.loc[mv_grid.index.difference(district_heating.index),:]
+        mv_grid_ind = mv_grid_ind.reset_index()
+        
+        CTS_demands_grid = pd.merge(demand,mv_grid_ind[['subst_id','zensus_population_id']],
+                                     on='zensus_population_id',
+                                     how='inner') 
+        
+        CTS_demands_grid.drop('zensus_population_id', axis =1, inplace =True)
+        CTS_demands_grid = CTS_demands_grid.groupby('subst_id').sum()
+        
+        CTS_per_grid =  pd.merge(CTS_per_grid,CTS_demands_grid[['demand']],
+                         how='inner', right_on =  CTS_per_grid.index, left_on = CTS_demands_grid.index)
+
+        CTS_per_grid =  CTS_per_grid.rename(columns={'key_0':'subst_id'})
+        CTS_per_grid.set_index('subst_id', inplace =True)
+        
+        CTS_per_grid =   CTS_per_grid[ CTS_per_grid.columns[:-1]].multiply( CTS_per_grid.demand,axis=0)
+        
+        CTS_per_zensus = 0
+
+    else:
+        CTS_per_district = 0
+        CTS_per_grid = 0
+        
+        CTS_per_zensus =  pd.merge(CTS_per_zensus,demand, how='inner',
+                                 right_on = CTS_per_zensus.index, left_on = demand.zensus_population_id)
+        CTS_per_zensus = CTS_per_zensus.drop('key_0',axis=1)
+        CTS_per_zensus.set_index('zensus_population_id',inplace =True)
+
+        CTS_per_zensus =  CTS_per_zensus[CTS_per_zensus.columns[:-1]].multiply(CTS_per_zensus.demand,axis=0)
+
+
+    return CTS_per_district, CTS_per_grid,  CTS_per_zensus
+
+
+def demand_profile_generator(aggregation_level = 'district'):
+    '''
+    
+    Description: Creating final demand profiles
+
+    Parameters
+    ----------
+    aggregation_level : str, optional
+        if further processing is to be done in zensus cell level 'other'
+        else 'dsitrict'. The default is 'district'.
+
+    Returns
+    -------
+    None.
+
+    '''
+    
+    residential_demand_dist, residential_demand_grid, residential_demand_zensus = residential_demand_scale(aggregation_level)
+    CTS_demand_dist, CTS_demand_grid, CTS_demand_zensus = CTS_demand_scale(aggregation_level)
+
+    
+    if aggregation_level =='district':
+        total_demands_dist = pd.concat([residential_demand_dist,CTS_demand_dist])
+        total_demands_dist.sort_index(inplace=True)
+        total_demands_dist = total_demands_dist.groupby(lambda x:x, axis=0).sum()
+        total_demands_dist.index.name = 'area_id'
+        
+        final_heat_profiles_dist = pd.DataFrame(index= total_demands_dist.index)
+        final_heat_profiles_dist['dist_aggregated_mw'] = total_demands_dist.values.tolist()
+        final_heat_profiles_dist.to_sql('egon_heat_time_series_dist',con=db.engine(),schema='demand' ,
```

Comment:
These are the time series for district heating and the ones in `egon_heat_time_series_grid` are for individual heating per mv grid? 
If yes, I would like to use table names which indicate this, e.g. `egon_timeseries_district_heating` and  `egon_timeseries_individual_heating`.

Reply:
When the aggregation is done per district level (default) two tables are written on the database. 'egon_heat_time_series_dist' and  'egon_heat_time_series_grid'. The first is the aggregation of all the cells falling within the district heating networks. The second table is all the remaining cells which do not fall under any district heating network and are thus aggregated per mv grid.

In case the aggregation is done per individual level, only one table is written in the database called 'egon_heat_time_series_zensus'. In this case every census cell will have a time series. (I have not run this scenario as the database is too big but should work.)

Reply:
Thank you for the explanation. Since all heat demands which are not part of a district heating grid are supplied by individual heat (e.g. a heat pump inside the house), these are the individual heat demand timeseries per mv grid used in eTraGo.
So maybe `egon_timeseries_district_heating` and `egon_etrago_timeseries_individual_heating` is better. The name of the table on zensus level is fine.

Reply:
If I understand correctly, you would just want the names of the table changed?

Reply:
Yes, I mean just the name of these two tables.

Reply:
Table names changed